internal/core/src/index/VectorMemIndex.cpp

```diff
@@ -521,6 +521,25 @@ VectorMemIndex<T>::Query(const DatasetPtr dataset,
 
     auto num_queries = dataset->GetRows();
     knowhere::Json search_conf = search_info.search_params_;
+    if (search_info.group_by_field_id_.has_value()) {
+        auto result = std::make_unique<SearchResult>();
+        try {
+            knowhere::expected<
+                std::vector<std::shared_ptr<knowhere::IndexNode::iterator>>>
+                iterators_val =
+                    index_.AnnIterator(*dataset, search_conf, bitset);
+            if (iterators_val.has_value()) {
```

Comment:
log error if the returned `knowhere::expected` has an non-ok code. I wonder is simply logging the error sufficient? should we wrap/return the error back to the client?

Reply:
changed